pkg/cmd/util/clientcmd/factory.go

```diff
@@ -106,6 +107,15 @@ func NewFactory(clientConfig kclientcmd.ClientConfig) *Factory {
 		return describe.NewHumanReadablePrinter(noHeaders), nil
 	}
 
+	generators := map[string]kubectl.Generator{
+		"route/v1": routegen.RouteGenerator{},
```

Comment:
@smarterclayton do you want `osc expose` to support exposing services from services or just routes?